test/src/test/java/jenkins/bugs/Jenkins64991Test.java

```diff
@@ -0,0 +1,163 @@
+
+        assertTrue(loginPage.isHtmlPage());
+        assertThat(loginPage.getUrl().toExternalForm(), containsStringIgnoringCase("%2Fmanage"));
+
+        ((HtmlTextInput)loginPage.getElementByName("j_username")).setText("alice");
+        ((HtmlPasswordInput)loginPage.getElementByName("j_password")).setText("alice");
+
+        final Page redirectedPage = HtmlFormUtil.submit(loginPage.getFormByName("login"));
+        assertTrue(redirectedPage.isHtmlPage());
+        assertEquals(j.getURL() + "manage", redirectedPage.getUrl().toExternalForm());
+        assertThat(redirectedPage.getWebResponse().getContentAsString(), containsStringIgnoringCase(Messages.GlobalSecurityConfiguration_DisplayName()));
+    }
+
+
+    @Test
+    public void testRedirect() throws Exception {
+        final JenkinsRule.WebClient webClient = j.createWebClient();
+        final HtmlPage indexPage = webClient.goTo("");
+
+        final Page loginPage = indexPage.getElementsByTagName("a").stream().filter(
+                e -> e.hasAttribute("href") && e.getAttribute("href").contains(j.jenkins.getSecurityRealm().getLoginUrl())
+        ).findFirst().orElseThrow(() -> new RuntimeException("cannot find login link")).click();
+        // Could be simplified to `indexPage.getElementById("login-link").click();` if we're willing to edit loginLink.jelly
+
+        assertTrue(loginPage.isHtmlPage());
+        assertThat(loginPage.getUrl().toExternalForm(), endsWithIgnoringCase("%2F"));
+
+        HtmlPage loginHtmlPage = (HtmlPage) loginPage;
+        ((HtmlTextInput)loginHtmlPage.getElementByName("j_username")).setText("alice");
+        ((HtmlPasswordInput)loginHtmlPage.getElementByName("j_password")).setText("alice");
+
+        final Page redirectedPage = HtmlFormUtil.submit(loginHtmlPage.getFormByName("login"));
+        assertTrue(redirectedPage.isHtmlPage());
+        assertEquals(indexPage.getUrl(), redirectedPage.getUrl());
+    }
+
+    @Test
+    public void testRedirectToProject() throws Exception {
+        final FreeStyleProject freeStyleProject = j.createFreeStyleProject();
+
+        final JenkinsRule.WebClient webClient = j.createWebClient();
+        final HtmlPage projectPage = webClient.getPage(freeStyleProject);
+        assertThat(projectPage.getWebResponse().getContentAsString(), containsStringIgnoringCase("Project " + freeStyleProject.getDisplayName()));
+
+        final Page loginPage = projectPage.getElementsByTagName("a").stream().filter(
+                e -> e.hasAttribute("href") && e.getAttribute("href").contains(j.jenkins.getSecurityRealm().getLoginUrl())
+        ).findFirst().orElseThrow(() -> new RuntimeException("cannot find login link")).click();
+        // Could be simplified to `projectPage.getElementById("login-link").click();` if we're willing to edit loginLink.jelly
+
+        assertTrue(loginPage.isHtmlPage());
+        assertThat(loginPage.getUrl().toExternalForm(), containsStringIgnoringCase("%2Fjob%2F" + freeStyleProject.getName()));
```

Comment:
Create a project with a fixed name so the assertion can be more concrete. (Preferably one with spaces so you can check `%20` escaping…)